lightning/src/ln/channel.rs

```diff
@@ -3509,26 +3501,144 @@ impl<Signer: Sign> Channel<Signer> {
+		} else {
+			// We got a reorg but not enough to trigger a force close, just update
+			// funding_tx_confirmed_in and return.
+			false
+		};
+
+		//TODO: Note that this must be a duplicate of the previous commitment point they sent us,
+		//as otherwise we will have a commitment transaction that they can't revoke (well, kinda,
+		//they can by sending two revoke_and_acks back-to-back, but not really). This appears to be
+		//a protocol oversight, but I assume I'm just missing something.
+		if need_commitment_update {
+			if self.channel_state & (ChannelState::MonitorUpdateFailed as u32) == 0 {
+				let next_per_commitment_point = self.holder_signer.get_per_commitment_point(self.cur_holder_commitment_transaction_number, &self.secp_ctx);
+				return Some(msgs::FundingLocked {
+					channel_id: self.channel_id,
+					next_per_commitment_point,
+				});
+			} else {
+				self.monitor_pending_funding_locked = true;
+			}
+		}
+		None
+	}
+
+	/// When a transaction is confirmed, we check whether it is or spends the funding transaction
+	/// In the first case, we store the confirmation height and calculating the short channel id.
+	/// In the second, we simply return an Err indicating we need to be force-closed now.
+	pub fn transactions_confirmed<L: Deref>(&mut self, block_hash: &BlockHash, height: u32, txdata: &TransactionData, logger: &L)
+			-> Result<Option<msgs::FundingLocked>, msgs::ErrorMessage> where L::Target: Logger {
+		let non_shutdown_state = self.channel_state & (!MULTI_STATE_FLAGS);
+		for &(index_in_block, tx) in txdata.iter() {
+			if let Some(funding_txo) = self.get_funding_txo() {
+				if non_shutdown_state & !(ChannelState::TheirFundingLocked as u32) == ChannelState::FundingSent as u32 {
+					if tx.txid() == funding_txo.txid {
+						let txo_idx = funding_txo.index as usize;
+						if txo_idx >= tx.output.len() || tx.output[txo_idx].script_pubkey != self.get_funding_redeemscript().to_v0_p2wsh() ||
+								tx.output[txo_idx].value != self.channel_value_satoshis {
+							if self.is_outbound() {
+								// If we generated the funding transaction and it doesn't match what it
+								// should, the client is really broken and we should just panic and
+								// tell them off. That said, because hash collisions happen with high
+								// probability in fuzztarget mode, if we're fuzzing we just close the
+								// channel and move on.
+								#[cfg(not(feature = "fuzztarget"))]
+								panic!("Client called ChannelManager::funding_transaction_generated with bogus transaction!");
+							}
+							self.channel_state = ChannelState::ShutdownComplete as u32;
+							self.update_time_counter += 1;
+							return Err(msgs::ErrorMessage {
+								channel_id: self.channel_id(),
+								data: "funding tx had wrong script/value".to_owned()
+							});
+						} else {
+							if self.is_outbound() {
+								for input in tx.input.iter() {
+									if input.witness.is_empty() {
```

Comment:
If we hit this branch, it does mean the confirmed transaction matches our expectation (`tx.txid() = funding_txo.index`) so we *might* be okay there, even if we're exposed to malleation during tx-relay.

That said our expected `funding_txo` might belong to a malleable transaction given as such by `get_outbound_funding_created` and in fact this method should recall that `funding_txo` MUST NOT belong to a malleable transaction.

Reply:
Yea, technically by then we're ok, but its also a critical bug in the implementation of the wallet calling LDK, so we should do *something*. Honestly, I think we should probably replace the whole FundingTransactionBroadcastable path to make the user provide us the transaction, but that's a separate PR.